servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java

```diff
@@ -1783,6 +1784,62 @@ public void testAssignRefToFreshMain() throws BaseNessieClientServerException {
     assertThat(testTagRef.getHash()).isEqualTo(main.getHash());
   }
 
+  @Test
+  public void testBranchesHaveMetadataProperties() throws BaseNessieClientServerException {
+    String branchPrefix = "branchesHaveMetadataProperties";
+    int numBranches = 5;
+    int commitsPerBranch = 10;
+
+    for (int i = 0; i < numBranches; i++) {
+      Reference r = api.createReference().reference(Branch.of(branchPrefix + i, null)).create();
+      String currentHash = r.getHash();
+      createCommits(r, 1, commitsPerBranch, currentHash);
+    }
+    // not fetching additional metadata
+    List<Reference> references = api.getAllReferences().get().getReferences();
+    Optional<Reference> main =
+        references.stream().filter(r -> r.getName().equals("main")).findFirst();
+    assertThat(main).isPresent();
+    List<Branch> branches =
+        references.stream()
+            .filter(r -> r.getName().startsWith(branchPrefix))
+            .map(r -> (Branch) r)
+            .collect(Collectors.toList());
+    assertThat(branches).hasSize(numBranches);
+    assertThat(branches).allSatisfy(branch -> assertThat(branch.metadataProperties()).isNull());
+
+    // fetching additional metadata for each branch
+    references = api.getAllReferences().fetchAdditionalMetadata().get().getReferences();
+    branches =
+        references.stream()
+            .filter(r -> r.getName().startsWith(branchPrefix))
+            .map(r -> (Branch) r)
+            .collect(Collectors.toList());
+    assertThat(branches).hasSize(numBranches);
+    assertThat(branches)
+        .allSatisfy(
+            branch -> {
+              Map<String, Object> metadataProperties = branch.metadataProperties();
+              assertThat(metadataProperties).isNotEmpty();
+              assertThat(metadataProperties)
+                  .hasFieldOrProperty("numCommitsAhead")
+                  .extracting("numCommitsAhead")
+                  .isEqualTo(commitsPerBranch);
+              assertThat(metadataProperties)
+                  .hasFieldOrProperty("numCommitsBehind")
+                  .extracting("numCommitsBehind")
+                  .isEqualTo(0);
+              assertThat(metadataProperties)
+                  .hasFieldOrProperty("headCommitMeta")
+                  .extracting("headCommitMeta")
+                  .isNotNull();
```

Comment:
Better assert the content of the `CommitMeta` here.

Reply:
done